cumulusci/tasks/salesforce/network_member_group.py

```diff
@@ -0,0 +1,164 @@
+from typing import List, Dict
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+from cumulusci.core.exceptions import SalesforceException, CumulusCIException
+from cumulusci.core.utils import process_list_arg
+
+
+class CreateNetworkMemberGroups(BaseSalesforceApiTask):
+    """
+    Creates NetworkMemberGroup for a Network (Experience Site) for Profiles and Permission Sets
+    that don't already have a corresponding NetworkMemberGroup.
+
+    Raises exceptions if records cannot be found:
+    - Network with Name network_name
+    - Profiles with Names in profile_names
+    - Permission Sets with Names in permission_set_names
+    """
+
+    task_options = {
+        "network_name": {
+            "description": (
+                "Name of Network to add NetworkMemberGroup children records."
+            ),
+            "required": True,
+        },
+        "profile_names": {
+            "description": (
+                "List of Profile Names to add as NetworkMemberGroups "
+                "for this Network."
+            ),
+            "required": False,
+        },
+        "permission_set_names": {
+            "description": (
+                "List of PermissionSet Names to add as NetworkMemberGroups "
+                "for this Network."
+            ),
+            "required": False,
+        },
+    }
+
+    def _get_network_id(self, network_name: str) -> str:
+        """
+        Returns Id of Network record with Name network_name.
+        Raises a SalesforceException if no Network is found.
+        """
+
+        networks = self.sf.query(
+            f"SELECT Id FROM Network WHERE Name = '{network_name}' LIMIT 1"
```

Comment:
Do we need to protect against SOQL injection?

Reply:
I don't think so. The operator of this automation already has access to execute any desired SOQL query against the target org.